### DISCUSSION TITLE
Is it possible to perform conditional post-render scripts depending on quarto call?

### DISCUSSION THREAD

**Discussion by liyongg** [quarto-dev/quarto-cli#6080] [Q&A]
### Description

For my Quarto book project, after each render I copy the rendered files to another directory: from `_book` to `output` in the root directory with some timestamps and additional information added. However, when using the preview function, the post-render script is executed as well and for every click on a chapter the directory is copied which is something I prefer not to have.

I would like to know if something like this would be possible in an R-script:

```r
if (call_is_quarto_render_and_not_quarto_preview()) {
  # run functions
}
```

This way, `quarto render` is the only way to trigger post-render scripts and `quarto preview` not trigger post-render functions.

**Comment by mcanouil** [comment]
See `QUARTO_PROJECT_RENDER_ALL ` described in <https://quarto.org/docs/projects/scripts.html#pre-and-post-render> which should address your needs.

Another possibility is to use project profile where in the default you don't have the post-render, see <https://quarto.org/docs/projects/profiles.html>.
